src/createServer.js

```diff
@@ -1,8 +1,105 @@
 'use strict';
 
+const url = require('url');
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer((req, res) => {
+    const normalizeUrl = new url.URL(req.url, `http://${req.headers.host}`);
+    const pathName = normalizeUrl.pathname;
+    const arrayFromPathName = pathName.slice(1).split('/');
+
+    if (pathName.includes('//')) {
+      res.writeHead(404, {
+        'Content-Type': 'text/plain',
+      });
+
+      res.end(
+        JSON.stringify({
+          error: 'Path to file should be like this /file/folderName/filename',
+        }),
+      );
+
+      return;
+    }
+
+    if (!pathName.startsWith('/file')) {
+      res.writeHead(400, {
```

Comment:
```suggestion
    if (!pathName.startsWith('/file/')) {
      res.writeHead(200, {
```
![image](https://github.com/user-attachments/assets/ff06c1c4-2fbd-418e-b704-3583e72eb1dc)